### ISSUE TITLE
Stipple multi-threading issues affecting REPL?

### ISSUE THREAD

**Issue by PallHaraldsson** [GenieFramework/Stipple.jl#23]
Hi, I chose Stipple (over Dash.jl) despite no tutorials, since I believe in you and your work.

I might have some questions if ok, but wanted to let you know of a potential issue.

See the unrelated questions in B.

The REPL issue is obvious, but strange when you see it, and a known issue, stuff can appear when it shouldn't. I'm just trying to figure out what triggers it, and it well may be Revise, not your code. I've had a similar issue before, then in Atom, now in VSCode, so I think I can rule those out.

I do not think it's my own code, that's very simple, some code I came up with modifying/trying to learn from your demo code. It only does "using Stipple, StippleCharts" and e.g. "[..] Stipple.register_components(Name, StippleCharts.COMPONENTS) [.. route and up]"

Would you think multi-threading is involved?

Everything is looks ok with running the code and stuff I do after in the REPL up to and including e.g. (but I quote all strangeness):
```
(@v1.5) pkg> up DataStructures
   Updating registry at `~/.julia/registries/General`
######################################################################## 100,0%
  Installed StatsBase ─ v0.33.3
  Installed Revise ──── v3.1.12
  Installed Genie ───── v1.15.0
No Changes to `~/.julia/environments/v1.5/Project.toml`
Updating `~/.julia/environments/v1.5/Manifest.toml`
  [c43c736e] ↑ Genie v1.14.0 ⇒ v1.15.0
  [295af30f] ↑ Revise v3.1.11 ⇒ v3.1.12
  [2913bbd2] ↑ StatsBase v0.33.2 ⇒ v0.33.3

[..]

julia> 27+56+59+48++32
ERROR: name not set for 5e69872b-a020-4c49-8103-8eebb826038c
Stacktrace:
ERROR:  [1] UndefVarError: ++ not definederror
Stacktrace:(^[[A^[[D
:: [1] Stringtop-level scope) at  at REPL[22]:1./error.jl:33

 [2] (::Revise.var"#40#41"{Dict{Base.PkgId,String},String})(::IOStream) at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/pkgs.jl:450
 [3] open(::Revise.var"#40#41"{Dict{Base.PkgId,String},String}, ::String; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./io.jl:325
 [4] open at ./io.jl:323 [inlined]
 [5] manifest_paths! at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/pkgs.jl:442 [inlined]
 [6] manifest_paths at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/pkgs.jl:485 [inlined]
 [7] (::Revise.var"#42#43"{String})() at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/pkgs.jl:500
 [8] with_logstate(::Function, ::Any) at ./logging.jl:408
 [9] with_logger at ./logging.jl:514 [inlined]
 [10] watch_manifest(::String) at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/pkgs.jl:497
 [11] (::Revise.TaskThunk)() at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/types.jl:272

julia> 27+56+59+48+32
ERROR: name not set for 5e69872b-a020-4c49-8103-8eebb826038c
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] (::Revise.var"#40#41"{Dict{Base.PkgId,String},String})(::IOStream) at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/pkgs.jl:450
 [3] open(::Revise.var"#40#41"{Dict{Base.PkgId,String},String}, ::String; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./io.jl:325
 [4] open at ./io.jl:323 [inlined]
 [5] manifest_paths! at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/pkgs.jl:442 [inlined]
 [6] manifest_paths at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/pkgs.jl:485 [inlined]
 [7] (::Revise.var"#42#43"{String})() at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/pkgs.jl:500
 [8] with_logstate(::Function, ::Any) at ./logging.jl:408
 [9] with_logger at ./logging.jl:514 [inlined]
 [10] watch_manifest(::String) at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/pkgs.jl:497
 [11] (::Revise.TaskThunk)() at /home/pharaldsson_sym/.julia/packages/Revise/moD4B/src/types.jl:272

julia> 27+56+59+48+32

julia> 27+56+59+48
222

julia> 27+56+59+48+32
222

julia> 27+56+59+48
190
```

B.
I did sign up for the beta, was e.g. intrigued by:

>Visual builder
Build pixel perfect apps.

I do have some 12 input fields that that are too wide, and split to the next line, but I would like narrower to fit the windows exactly or some exact size. Such is possible in HTML (or rather CSS), and I assume Vue (and all similar, also in Dash.jl/React), it's just that I'm no expert on it. I did find a builder for vue, or was it quasar? Is that what you had in mind? Is my impression correct that this is a forward-looking statement that you can use a visual builder with Stipple (since you do not point too), or can you and should use already available non-Stipple tools with it? While I shouldn't need to do CSS (usually) with, could I still add CSS if I wanted to, or would I be overlooking some other way to do CSS-like functionality?

**Comment by essenciary** [comment]
@PallHaraldsson Thanks so much - yes, the lack of docs is a big issue. Atm working on a couple more key features to stabilize the API then we'll focus on the docs and tests for a while. This is also holding back on the UI builder work, which will come after. It is a dedicated UI builder for Stipple apps based on Grapes.js. Yes, it is/will be possible to add CSS and one key aspect is themes support, allowing to replace the whole look and feel of an app just by installing/swapping pre-made themes. 

I'm not sure about the REPL issue. Is there a Manifest file with a UUID of `5e69872b-a020-4c49-8103-8eebb826038c`. What does the UUID point to?

**Comment by PallHaraldsson** [comment]
There's no Manifest file. I'm just using your modified code, but there is (not something I'm using, directly; nor indirectly?):
  [5e69872b] Xcb.jl v0.2.0 `https://gitlab.com/poposca/Xcb.jl.git#master`

**Comment by PallHaraldsson** [comment]
>It is a dedicated UI builder for Stipple apps based on Grapes.js.

Good to know (is that a replacement for Quasar.dev? It's not obviously used together, seems like competitors). I took a quick look, it e.g. supports mobile. I know web-apps should be "mobile-first", for me not really a concern. But good to know supports the whole range. I did see (additionally) for Quasar, desktop apps possible (i.e. Electron app), at:

https://quasar.dev/introduction-to-quasar

I'm personally ok with web page only for now, but does this mean Stipple.jl would support Electron? Something on the horizon, or at least doesn't contradict (should an add-on package provide that capability?).

**Comment by PallHaraldsson** [comment]
FYI: The REPL issue is also there in Julia 1.6-rc1

![image](https://user-images.githubusercontent.com/8005416/109033113-36315a80-76be-11eb-83b2-1d84a499dea3.png)

If I copy paste the invisible stuff I get:

```
R: Unsatisfiable requirements detected for package Stipple [4acbeb90]:
 Stipple [4acbeb90] log:
 ├─possible versions are: 0.1.0-0.9.0 or uninstalled
 ├─restricted to versions [0.5-0.6, 0.8-0.9] by StippleUI [a3c5d34a], leaving only versions [0.5.0-0.6.0, 0.8.0-0.9.0]
 │ └─StippleUI [a3c5d34a] log:
 │   ├─possible versions are: 0.2.4 or uninstalled
 │   └─StippleUI [a3c5d34a] is fixed to version 0.2.4
 └─restricted to versions 0.2.0 by an explicit requirement — no versions left
```

This is clearly a bug in Julia not Stipple, but may be triggered by Stipple (threading).
